server/src/main/java/org/opensearch/indices/IndicesRequestCache.java

```diff
@@ -491,7 +513,7 @@ private void updateCleanupKeyToCountMapOnCacheInsertion(CleanupKey cleanupKey) {
         }
 
         private void updateCleanupKeyToCountMapOnCacheEviction(CleanupKey cleanupKey) {
-            if (stalenessThreshold == 0.0 || cleanupKey.entity == null) {
+            if (cleanupKey.entity == null) {
```

Comment:
cleanUpKey.entity can be null for older shard cases(due to recent changes) and I think we will end up skipping decrementing stale counter? Can be verified again via an IT test.

Reply:
we discussed the entity will be null for a deleted Index but not for closed index since shard will still be lying around in that case. 
We have a fix for deleted index and when entity is null.

Reply:
Added ITs for deleted and closed scenarios